### DISCUSSION TITLE
Add a seal of approval for papers-related implementation ?

### DISCUSSION THREAD

**Discussion by QuanticDisaster** [pyg-team/pytorch_geometric#3395] [Ideas]
Behind the pompuous title, what I am thinking is to propose an addition to documentations such as "validated/reviewed by original author". This could appear on convolutions (torch.nn.conv), or entire models (PyG examples on classif/segmentation) and (optionally) training and preprocessing parameters for datasets.

The idea is to have additional guarantees on different elements and to have authors actively participate in PyG without taking too much of their time. I think it would be also be beneficial in making PyG more popular, especially when most models are stuck with a fixed number of points in PyTorch, or with sampling strategies herited from PointNet which are now realized in few lines thanks to PyG utils elements

**Comment by rusty1s** [comment]
This is an interesting idea, although probably hard to realize in practice as we need to reach out to authors and authors need to find time to review. It's also hard to always guarantee full reproducibility due to minor differences in dataset processing, implementation, etc. In the end, we are trying to reproduce models and layers in a unified way and in a single example file, which sometimes trade performance for simplicity of the example.

**Reply by QuanticDisaster** [reply]
I agree, this is why full reproducibility should not be the goal here and we could brush it off to torch-points3D at least for the moment :  https://github.com/nicolas-chaulet/torch-points3d

> In the end, we are trying to reproduce models and layers in a unified way and in a single example file, which sometimes trade performance for simplicity of the example.

You are right it should stay that way, most of these models have a github repo where they are fully implemented by their author but are sometimes difficult to grasp. It is just to have some "validation" on the author part to check details of convolutions operations, or intermediate layers dimensions when they have spare time

We could begin with an award to this guy :p : https://github.com/pyg-team/pytorch_geometric/discussions/3129 🥇
